ai2thor/fifo_server.py

```diff
@@ -93,19 +101,52 @@ def __init__(
         }
 
         self.eom_header = self._create_header(FieldType.END_OF_MESSAGE, b"")
-        super().__init__(width, height, depth_format, add_depth_noise)
+        super().__init__(
+            width=width,
+            height=height,
+            timeout=timeout,
+            depth_format=depth_format,
+            add_depth_noise=add_depth_noise
+        )
 
     def _create_header(self, message_type, body):
         return struct.pack(self.header_format, message_type, len(body))
 
-    def _recv_message(self):
+    def _read_with_timeout(self, server_pipe, message_size: int, timeout: Optional[float]):
+        if timeout is None:
+            return server_pipe.read(message_size)
+
+        start_t = time.time()
+        message = b""
+
+        while message_size > 0:
```

Comment:
Is this way of reading with the while loop and directly from the file descriptor any slower? I don't think it should but an option could be if `server_timeout` is `None` then we could use the old function.

Reply:
That's already there if look at lines 116-117.